### DISCUSSION TITLE
VCPKG Asset Caching for company internal assets

### DISCUSSION THREAD

**Discussion by mattias-karlsson** [microsoft/vcpkg#42791] [Q&A] [answered]
Hi,

I use the Microsoft official VCPKG registry as my default registry and a company internal VCPKG registry for our internal ports/libraries.

A vcpkg-configuration.json could typically look like this:

```
{
  "default-registry": {
    "kind": "git",
    "baseline": "84a143e4caf6b70db57f28d04c41df4a85c480fa",
    "repository": "https://github.com/microsoft/vcpkg"
  },
  "registries": [
    {
      "kind": "git",
      "repository": "https://git.company.com/company/company-vcpkg-registry",
      "baseline": "5d647865aba3571483c84ca3bd9b62db2966b4e3",
      "packages": [
        "company-*"
      ]
    }
  ]
}
```
All ports except those prefixed with "company-" will be fetched from the official registry. 

I have started to try out VCPKG asset caching and it works perfectly! But I have noticed that VCPKG only caches the external assets and no assets from our company internal git repositories pointed out by the ports in our internal registry is pushed to the cache. That is perfectly fine because we are in control of the internal assets but I see nothing in the documentation about this. I just wonder if someone can confirm this behavior.

So, is it by design to not put our company internal assets in the asset cache, and if so, how does VCPKG make the difference?

**Comment by Neumann-A** [comment]
Are you using vcpkg_from_git in your internal ports?

**Reply by mattias-karlsson** [reply]
Yes.

**Reply by Neumann-A** [reply, marked as answer]
Then that is your answer. vcpkg_from_git is not part of the asset caching. There is no SHA512 which could be used.

**Reply by mattias-karlsson** [reply]
Ah, of course. Makes perfect sense. Thanks!
